### DISCUSSION TITLE
Command palette shortcut collides with common browser shortcut

### DISCUSSION THREAD

**Discussion by brylie** [community/community#15255] [Other Feature Feedback, Questions, & Ideas] [answered]
The command palette shortcut, `CTRL+K` collides with a common browser shortcut. Normally, pressing `CTRL+K` will let the user input text into the browser search/URL bar (on Chrome and Firefox).

Please consider existing keyboard shortcuts and their uses when designing keyboard shortcuts for GitHub, as browser/OS functionality should take precedent over the GitHub commands.

**Comment by iensu** [comment]
Agree with this. The best would be for the user to be able to map keyboard shortcuts as they see fit.

**Comment by queenofcorgis** [comment, marked as answer]
Hi there @brylie 👋🏼  Thanks for the feedback, we have a feature that allows you to customize the keyboard shortcuts used to open the command palette. You can edit your shortcuts in the [Accessibility section](https://github.com/settings/accessibility) of your user settings.

**Reply by craigglennie** [reply]
@liligalante they're not asking about the shortcut to open the command palette, they're asking for Github to stop overriding existing browser/OS functionality by hijacking various keyboard shortcuts. Or to give users a way to opt in/out or to customise.

Personally I find hijacking Cmd+F for find on Mac to be absolutely obnoxious and I can't believe I can't disable it.

**Reply by PawelWMS** [reply]
The command palette feature also seems to be unable to recognize the difference between `ctrl + f` and `ctrl + shift + f`, so it hijacks even more shortcuts.

**Reply by bhrgunatha** [reply]
CTRL+K is a built-in shortcut used by both Firefox and Chrome.  

Please change the default [so it doesn't override them.](https://support.mozilla.org/en-US/kb/keyboard-shortcuts-perform-firefox-tasks-quickly).  It's obnoxious behaviour for a website. 

Imagine if some website sent you their home page whenever you hit "H".

**Reply by cwendling** [reply]
…and given the number of recent tickets about this, it's not just a couple of weirdos either :)

**Reply by rsherkp4** [reply]
Would you just fix this. Some of us have ctrl-k burned into muscle memory. If I want to search for something on Github or some other site, I'll just click on your search box.
I would also assume the only people who will discover this feature are the ones hitting the ctrl-k combo to get a BROWSER search.

**Reply by yermulnik** [reply]
> we have a feature that allows you to customize the keyboard shortcuts used to open the command palette. You can edit your shortcuts in the [Accessibility section](https://github.com/settings/accessibility) of your user settings.

Has this feature been removed? I stumbled upon ctrl-k clash today and can't seem to see the option to customize gh command palette keyboard shortcut at [Accessibility section](https://github.com/settings/accessibility) 🤔

**Reply by yermulnik** [reply]
Ah, command palette feature preview somehow got disabled on its own. Re-enabling it fixed the issue.

**Reply by Loque18** [reply]
> Ah, command palette feature preview somehow got disabled on its own. Re-enabling it fixed the issue.

same for me thanks a lot man !!

**Comment by cwendling** [comment]
It's nice that it's customizable, but having it default to a standard browser shortcut as common as focusing the search field is not good UX design for anyone.

I have been annoyed by this a couple times and *never* meant to open the command palette, so this was just GitHub getting in the way; using a less used shortcut by default would be a *lot* better.

Even the available customization are questionable:
* <kbd>Control+k</kbd>: focus search field in Firefox (possibly others, as it's a common shortcut in other apps as well)
* <kbd>Control+Shift+k</kbd>: open developer console in Firefox
* <kbd>Control+p</kbd>: print page (Firefox and many others).  It's admittedly uncommon to want to print out a GitHub page, but it's still a widespread shortcut used for something else, thus a hard habit to get.
* <kbd>Control+Shift+p</kbd>: open private browsing window in Firefox

This leaves us with only <kbd>Control+Alt+k</kbd> in Firefox if we don't want to interfere; and that means there's no way to bind both _Search mode_ and _Command mode_ without interfering.

IMO, there should be more customization options, and the default should be more carefully selected not to interfere with any reasonable shortcut from any common web browser.

**Comment by hinell** [comment]
It would be much nicer if chrome allowed to remap shortcuts bound to letters. I don't like the way it's hardwired.

**Comment by graciejiujitsu** [comment]
I agree that relying on users to remap defaults isn't an ideal long-term UX solution, especially when common browser habits are involved. Similar usability trade-offs come up in tools like [virtual assistant lead generation services](https://pearllemonleads.nl/services/lead-generation-for-virtual-assistants/), where familiar workflows usually reduce friction rather than requiring extra configuration.
